src/service-metrics.ts

```diff
@@ -3,55 +3,137 @@
 
 import { MeterProvider, PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics'
 import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-http'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import { BasicTracerProvider, TraceIdRatioBasedSampler,
+         ParentBasedSampler, BatchSpanProcessor } from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { Resource } from '@opentelemetry/resources'
+import {trace} from '@opentelemetry/api'
+
 import { Utils } from './utils.js'
 
 export const UNKNOWN_CALLER = 'Unknown'
 
 export const CONCURRENCY_LIMIT = 1
+export const TRACE_CONCURRENCY_LIMIT = 1
+export const DEFAULT_TRACE_SAMPLE_RATIO = 0.1
+
+enum LOG_LEVEL {
+    debug,
+    info,
+    warn,
+    imp,
+    error
+}
+
+class _NullLogger {
```

Comment:
Why leading underscores here?

Reply:
this is an internal method only - i can remove